### PR TITLE
[AndroidCrypto] Avoid printing JNI exceptions in cases where they are expected

### PR DESCRIPTION
**Pull request by elinor-fung**
`AndroidCryptoNative_X509GetContentType` relies on errors to determine what a byte array represents. Avoid the noise of printing the exceptions in this case.

cc @jkoritzinsky @steveisok @AaronRobinsonMSFT @bartonjs

### CODE REVIEW

**Comment by elinor-fung** [comment]
@jkoritzinsky It seems like we're just ignoring exceptions from PrivateKey.destroy - is an exception there sometimes expected and something we should stop printing as an error?
 https://github.com/dotnet/runtime/blob/6d238511167bdbf8cd16103e84b6befb45c0e301/src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_eckey.c#L50-L52

**Comment by jkoritzinsky** [comment]
Yes, I believe those exceptions are expected. They'd be a great use case for clearing the error and not printing it.

**Review by jkoritzinsky — APPROVED** [review]

File: src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c [hunk 1]
```diff
@@ -331,7 +331,7 @@ static jobject AndroidCryptoNative_CreateKeyPairFromCurveParameters(
     {
         // Destroy the private key data.
         (*env)->CallVoidMethod(env, loc[privateKey], g_destroy);
-        CheckJNIExceptions(env); // The destroy call might throw an exception. Clear the exception state.
+        TryClearJNIExceptions(env); // The destroy call might throw an exception. Clear the exception state.
```
Comment (AaronRobinsonMSFT) [inline src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c:334]: When intentionally ignoring return values I find it helpful to cast the return to `(void)` so it is clear the return is intentionally ignored. For example,

```c
(void)TryClearJNIExceptions(env);
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by bartonjs — APPROVED** [review]

**Comment by elinor-fung** [comment]
Timeout in staging is unrelated

**Pull request merged**